engine/src/engine.rs

```diff
@@ -816,10 +816,9 @@ impl<'env, I: IO + Copy, E: Env, M: ModExpAlgorithm> Engine<'env, I, E, M> {
                     gas_used: submit_result.gas_used,
                 }),
             })
-            .map_err(|e| {
-                sdk::log!("{:?}", e);
+            .inspect_err(|_e| {
+                sdk::log!("{:?}", _e);
```

Comment:
```suggestion
            .inspect_err(|e| {
                sdk::log!("{:?}", e);
```

Reply:
The `_e` variable couldn't be used if the feature `log` is not provided.